device/ufispace/x86_64-ufispace_s9300_32d-r0/pddf/pd-plugin.json

```diff
@@ -0,0 +1,98 @@
+{
+    
+    "XCVR":
+    {
+        "xcvr_present":
+        {
+           "i2c":
+           {
+               "valmap-SFP": {"1":true, "0":false },
+               "valmap-QSFP-DD": {"1":true, "0":false}
+           }
+
+        },
+        "plug_status":
+        {
+            "inserted": "1",
+            "removed": "0"
+        }
+    },
+    "PSU":
+    {
+        "psu_present": 
+        {
+            "i2c":
+            {
+                "valmap": { "1":true, "0":false }
+            },
+            "bmc":
+            {
+                "valmap": { "0x0280|":true, "0x0180|":false }
+            }
+        }, 
+		
+        "psu_power_good": 
+        {
+            "i2c":
+            {
+                "valmap": { "1": true, "0":false }
+            },
+            "bmc":
+            {
+                "valmap": { "0x0280|":true, "0x0180|":false }
+            }        
+        },
+
+        "psu_fan_dir":
+        {
+            "i2c":
+            {
+                "valmap": { "F2B":"EXHAUST", "B2F":"INTAKE" }
+            }
+        },
+
+        "PSU_FAN_MAX_SPEED":"30000"
+    },
+
+    "FAN":
+    {
+        "direction":
+        {
+            "bmc":
+            {
+                "valmap": {"0": "UNKNOW", "1":"INTAKE", "2":"EXHAUST"}
+            }
+        },
+
+        "present":
+        {
+            "i2c":
+            {
+                "valmap": {"1":true, "0":false}
+            },
+            "bmc":
+            {
+                "valmap": { "0x0280|":true, "0x0180|":false }
+            }      
+        },
+        
+        "duty_cycle_to_pwm": "lambda dc: ((dc*100)/625 - 1)",
+        "pwm_to_duty_cycle": "lambda pwm: (((pwm+1)*625)/100)",
+        "FAN_R_MAX_SPEED":"32000",
+        "FAN_F_MAX_SPEED":"36200"
+    },
+
+    "LED":
+    {
+        "capability":
+        {
+            "ro": ["SYS_LED", "FAN_LED", "PSU1_LED", "PSU2_LED"],
```

Comment:
This 'rw' or 'ro' flags are supported in pddf-device.json structure now. Please check this example below,

    "LOC_LED":
    {
        "dev_info": { "device_type":"LED", "device_name":"LOC_LED"},
        "dev_attr": { "index":"0", **"flag": "rw"**},
        "i2c" :
        {
            "attr_list":
            [
                {"attr_name":"blue_blink", "descr": "" , "bits" : "5:0",  "value" : "0x27", "swpld_addr" : "0x60", "swpld_addr_offset" : "0x25"},


You can make use of this provision.

Reply:
Thanks for your kindly notification, will update this